HabitLogger/Program.cs

```diff
@@ -0,0 +1,215 @@
+﻿using System.Text;
+using HabitLogger;
+using HabitLoggerLibrary;
+
+RunApplication();
+
+static void RunHabitLogger()
+{
+    using var connection = new HabitLoggerConnection();
+    bool isDone = false;
+    while (!isDone)
+    {
+        DisplayMenu();
+        int option = ParseOption();
+        bool continueApplication = ExecuteOption(option, connection);
+
+        isDone = !continueApplication;
+    }
+}
+
+static bool ExecuteOption(int option, HabitLoggerConnection connection)
+{
+    if (connection == null)
+    {
+        return false;
+    }
+
+    switch (option)
+    {
+        case (int)Options.ExitApplication:
+            return false;
+        case (int)Options.ReadLogs:
+            ReadLogs(connection);
+            break;
+        case (int)Options.AddNewLog:
+            AddNewLog(connection);
+            break;
+        case (int)Options.DeleteLog:
+            DeleteLog(connection);
+            break;
+        case (int)Options.UpdateLog:
+            UpdateLog(connection);
+            break;
+        default:
+            Console.WriteLine("Invalid option.");
+            Console.WriteLine("Must choose one of the options given.");
+            break;
+    }
+    return true;
+}
+
+static void UpdateLog(HabitLoggerConnection connection)
+{
+    Console.WriteLine("Updating log...");
+    Console.WriteLine();
+    int id = GetId();
+    int cupsOfWater = GetCupsOfWater();
+    bool isSuccess = connection.UpdateLog(id, cupsOfWater);
+
+    if (isSuccess)
+    {
+        Console.WriteLine($"Updated log of id {id} with {cupsOfWater} cups of water.");
+    } else
+    {
+        Console.WriteLine("Update was not successful.");
+    }
+    Console.WriteLine();
+}
+
+static void DeleteLog(HabitLoggerConnection connection)
+{
+    Console.WriteLine("Delete log...");
+    Console.WriteLine();
+    int id = GetId();
+    bool isSuccess = connection.DeleteLog(id);
+    if (isSuccess)
+    {
+        Console.WriteLine($"Log of id {id} was successfully deleted.");
+    } else
+    {
+        Console.WriteLine($"Unable to delete a log of id {id}.");
+    }
+
+    Console.WriteLine();
+}
+
+static int GetId()
+{
+    Console.Write("Type id of the log: ");
+    string? input = Console.ReadLine();
+    int id;
+
+    while (string.IsNullOrEmpty(input) || !int.TryParse(input, out id))
+    {
+        Console.Write("Invalid input. Type a valid id: ");
+        input = Console.ReadLine();
+    }
+    return id;
+}
+
+static void AddNewLog(HabitLoggerConnection connection)
```

Comment:
It would be good to let the user add the date for the log as well. Let's say I didn't have access to my computer for a day. I wouldn't be able to create a log for that day. What do you think?

Reply:
It makes sense to allow users the ability to provide their date. It more flexible for the user! At the same time, I believe my approach of just logging the current date is a good in the sense that it's easier for users to add a new log since they don't need to add their date. This makes it quicker and easier for the users. At the end though, both approaches have their pros and cons and are good!

I updated the add new log function so that the user will have to provide a date.